jscomp/build_tests/cli_help/input.js

```diff
@@ -0,0 +1,243 @@
+  "Options:\n" +
+  "  -v, -version  display version number\n" +
+  "  -h, -help     display help\n" +
+  "\n" +
+  "Subcommands:\n" +
+  "  build\n" +
+  "  clean\n" +
+  "  format\n" +
+  "  convert\n" +
+  "  dump\n" +
+  "  help\n" +
+  "\n" +
+  "Run `rescript <subcommand> -h` for subcommand help. Examples:\n" +
+  "  rescript build -h\n" +
+  "  rescript format -h\n";
+
+const buildHelp =
+  "Usage: rescript build <options> -- <ninja_options>\n" +
+  "\n" +
+  "`rescript build` builds the project with dependencies\n" +
+  "\n" +
+  "`rescript -- -h` for Ninja options (internal usage only; unstable)\n" +
+  "\n" +
+  "Options:\n" +
+  "  -w          Watch mode\n" +
+  "  -ws         [host]:port set up host & port for WebSocket build notifications\n" +
+  "  -verbose    Set the output to be verbose\n" +
+  "  -with-deps  *deprecated* This is the default behavior now. This option will be removed in a future release\n";
+
+const cleanHelp =
+  "Usage: rescript clean <options>\n" +
+  "\n" +
+  "`rescript clean` cleans build artifacts\n" +
+  "\n" +
+  "Options:\n" +
+  "  -verbose    Set the output to be verbose\n" +
+  "  -with-deps  *deprecated* This is the default behavior now. This option will be removed in a future release\n";
+
+const formatHelp =
+  "Usage: rescript format <options> [files]\n" +
+  "\n" +
+  "`rescript format` formats the current directory\n" +
+  "\n" +
+  "Options:\n" +
+  "  -stdin  [.res|.resi|.ml|.mli] Read the code from stdin and print\n" +
+  "          the formatted code to stdout in ReScript syntax\n" +
+  "  -all    Format the whole project \n" +
+  "  -check  Check formatting for file or the whole project. Use `-all` to check the whole project\n";
+
+const convertHelp =
+  "Usage: rescript convert <options> [files]\n" +
+  "\n" +
+  "`rescript convert` converts the current directory\n" +
+  "\n" +
+  "**This command removes old OCaml files and creates new ReScript \n" +
+  "files. Make sure your work is saved first!**\n" +
+  "\n" +
+  "Options:\n" +
+  "  -all  Convert the whole project\n";
+
+const dumpHelp =
+  "Usage: rescript dump <options> [target]\n" +
+  "`rescript dump` dumps the information for the target\n";
+
+// Shows build help with --help arg
+let out = child_process.spawnSync(`../../../rescript`, ["build", "--help"], {
+  encoding: "utf8",
+  cwd: __dirname,
+});
+assert.equal(out.stdout, buildHelp);
+assert.equal(out.stderr, "");
+assert.equal(out.status, 0);
+
+// FIXME: Help works incorrectly in watch mode
+out = child_process.spawnSync(`../../../rescript`, ["build", "-w", "--help"], {
+  encoding: "utf8",
+  cwd: __dirname,
+});
+// FIXME: Shouldn't have "Start compiling" for help
+assert.equal(out.stdout, ">>>> Start compiling\n" + buildHelp);
+// FIXME: Don't run the watcher when showing help
+assert.match(
+  out.stderr,
+  new RegExp(
+    "Uncaught Exception Error: ENOENT: no such file or directory, watch 'bsconfig.json'\n"
+  )
+);
+// FIXME: Should be 0
+assert.equal(out.status, 1);
```

Comment:
It'll be only possible to fix by moving the args parsing logic in one place. I want to do it when I have more free time.

Reply:
What about the other three FIXMEs above? Also something for later?

Reply:
Yeah, it's all in one